### ISSUE TITLE
Weird List Parsing

### ISSUE THREAD

**Issue by theDragonFire** [raphlinus/pulldown-cmark#149]
While using a wrapper for pulldown-cmark, [cmark](https://github.com/FuGangqiang/cmark.rs), I ran into what seems to me to be an error.

I made a nested list in the following form:
```
* List
  1. Sublist
    * Stuff
    * Stuff
  2. Sublist
    * Stuff
```
and the output (with added indentation) was :
```
<ul>
  <li>List
    <ol>
      <li>Sublist</li>
    </ol>
    <ul>
      <li>Stuff</li>
      <li>Stuff</li>
    </ul>
    <ol start="2">
      <li>Sublist</li>
    </ol>
    <ul>
      <li>Stuff</li>
    </ul>
  </li>
</ul>
```
while what I would expect it to be is:
```
<ul>
  <li>List
    <ol>
      <li>Sublist
        <ul>
          <li>Stuff</li>
          <li>Stuff</li>
        </ul>
      </li>
      <li>Sublist</li>
        <ul>
          <li>Stuff</li>
        </ul>
      </li>
  </li>
</ul>
```

If I am mistaken in my expectations, please correct me.

**Comment by raphlinus** [comment]
Confirmed, this is a bug. I wonder why it's taken this long to surface. Thanks for the report.

**Comment by marcusklaas** [comment]
Hmmm, it seems we may have regressed here. Current master produces once again
```html
<ul>
  <li>List
    <ol>
      <li>Sublist</li>
    </ol>
    <ul>
      <li>Stuff</li>
      <li>Stuff</li>
    </ul>
    <ol start="2">
      <li>Sublist</li>
    </ol>
    <ul>
      <li>Stuff</li>
    </ul>
  </li>
</ul>
```
I can investigate.

**Comment by raphlinus** [comment]
I thought I tried this, but as we've found, lists are tricky.

**Comment by marcusklaas** [comment]
Ok, I investigated and I think I found out what's going on here! We're actually spec compliant since the indentation for numbered lists is the length of the numeral, dot and following whitespace combined. See this example in the spec: https://spec.commonmark.org/0.28/#example-234

In the above snippet, the deepest sublists only have 2 additional spaces of indentation, where 3 is required. This modified example does produce the output you'd expect:
```md
* List
  1. Sublist
     * Stuff
     * Stuff
  2. Sublist
    * Stuff
```
